src/components/InsightWidget.tsx

```
@@ -430,27 +468,279 @@ export const InsightWidget: React.FC<InsightWidgetProps> = ({ className, onRemov
+    // Make sure the digest at this index exists
+    if (!allDigests[newIndex]) return;
+    
+    if (newIndex !== selectedIndex) {
+      // First, animate the current content out if ref exists
+      if (contentAnimeRef.current) {
+        // Store the current content height to prevent layout jumps
+        const currentHeight = contentAnimeRef.current.clientHeight;
+        contentAnimeRef.current.style.height = `${currentHeight}px`;
+        
+        // Stop any ongoing animations
+        if (animeScope.current && animeScope.current.remove) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)